### PR TITLE
RSDK-2607 add support for xarm lite

### PR DESCRIPTION
**Pull request by biotinker**
Adds support for the xarm lite arm

### CODE REVIEW

File: components/arm/fake/fake.go [hunk 1]
```diff
@@ -37,9 +37,9 @@ type AttrConfig struct {
 func modelFromName(model, name string) (referenceframe.Model, error) {
 	switch resource.ModelName(model) {
 	case xarm.ModelName6DOF.Name:
```
Comment (raybjork) [inline components/arm/fake/fake.go:39]: ` case xarm.ModelName6DOF.Name, xarm.ModelName7DOF.Name`

File: components/arm/xarm/xarm.go [hunk 2]
```diff
@@ -128,11 +150,13 @@ func NewxArm(ctx context.Context, cfg config.Component, logger golog.Logger, dof
 		return nil, err
 	}
 
-	model, err := Model(cfg.Name, dof)
+	model, err := Model(cfg.Name, modelName)
 	if err != nil {
 		return nil, err
 	}
 
+	dof := len(model.DoF())
```
Comment (raybjork) [inline components/arm/xarm/xarm.go:158]: can put this on L161

File: components/arm/fake/fake.go (hunk 1 above)
Comment (biotinker) [reply components/arm/fake/fake.go:39]: Good catch, plus Lite also needs to be added

File: components/arm/xarm/xarm.go [hunk 3]
```diff
@@ -103,10 +111,24 @@ func init() {
 		},
 		&AttrConfig{},
 	)
+
+	// xArmLite
+	registry.RegisterComponent(arm.Subtype, ModelNameLite, registry.Component{
```
[the remark this answers is not in the document]
Comment (biotinker) [reply components/arm/xarm/xarm.go:116]: Agree

File: components/arm/xarm/xarm.go [hunk 4]
```diff
@@ -56,18 +56,26 @@ var xArm6modeljson []byte
...
+var xArmLitemodeljson []byte
+
 // ModelName6DOF is a function used to get the string used to refer to the xarm model of 6 dof.
 var ModelName6DOF = resource.NewDefaultModel("xArm6")
 
 // ModelName7DOF is a function used to get the string used to refer to the xarm model of 7 dof.
 var ModelName7DOF = resource.NewDefaultModel("xArm7")
 
+// ModelNameLite is a function used to get the string used to refer to the xarm lite model of 6 dof.
+var ModelNameLite = resource.NewDefaultModel("xArmLite")
```
[the remark this answers is not in the document]
Comment (biotinker) [reply components/arm/xarm/xarm.go]: Updated to const strings; take a look

**Review by raybjork — APPROVED** [review]

File: components/arm/xarm/xarm.go [hunk 5]
```diff
@@ -56,57 +56,50 @@ var xArm6modeljson []byte
...
+	switch modelName {
+	case ModelName6DOF:
 		return referenceframe.UnmarshalModelJSON(xArm6modeljson, name)
-	case 7:
+	case ModelNameLite:
+		return referenceframe.UnmarshalModelJSON(xArmLitemodeljson, name)
+	case ModelName7DOF:
 		return referenceframe.UnmarshalModelJSON(xArm7modeljson, name)
 	default:
 		return nil, errors.New("no kinematics model for xarm with specified degrees of freedom")
```
Comment (spieswl) [inline components/arm/xarm/xarm.go:78]: [nit] Would prefer this error be updated to track the change to switching based on a model identifier and not arm DoF

**Review by spieswl — CHANGES_REQUESTED** [review]
Overall simplification is really nice, just one small nit on the `Model` switch case.

File: components/arm/xarm/xarm.go (hunk 5 above)
Comment (biotinker) [reply components/arm/xarm/xarm.go]: Updated

**Review by spieswl — APPROVED** [review]
Excellent, LGTM

**Pull request merged**